lua/lspconfig/configs.lua

```diff
@@ -72,17 +72,10 @@ function configs.__newindex(t, config_name, config_def)
     end
 
     if config.autostart == true then
-      local event
-      local pattern
-      if config.filetypes then
-        event = 'FileType'
-        pattern = table.concat(config.filetypes, ',')
-      else
-        event = 'BufReadPost'
-        pattern = '*'
-      end
-      api.nvim_create_autocmd(event, {
-        pattern = pattern,
+      local event_conf = config.filetypes and { event = 'FileType', pattern = table.concat(config.filetypes, ',') }
```

Comment:
`pattern` can be a table
```suggestion
      local event_conf = config.filetypes and { event = 'FileType', pattern = config.filetypes }
```